doc/whats-new.rst

```diff
@@ -20,90 +20,100 @@ What's New
 v2024.02.0 (unreleased)
 -----------------------
 
+This release brings size information to the text ``repr``, changes to the accepted frequency
+strings, and various bug fixes.
```

Comment:
I'm not too confident about the summary of the frequency string changes. Also, should we include `flox` here? Am I missing anything else that's important? And since the `datatree` changes are not supposed to be user-facing yet I also didn't include that.

Reply:
I think this is fine. Thanks!